### PR TITLE
update polkadot-sdk

### PR DESCRIPTION
**Pull request by xlc**
blocked by https://github.com/paritytech/polkadot-sdk/issues/5210 so we don't need to deal with node changes

the client node part is moved to https://github.com/AcalaNetwork/acala-node

#2745 #2596 to finish the Makefile and CI integration

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -39,68 +39,29 @@ build-full: githooks
...
 build-all:
-	cargo build --locked --features with-all-runtime
+	cargo build --locked
 
 .PHONY: build-benches
 build-benches:
 	cargo bench --locked --no-run --features wasm-bench --package module-evm
 	cargo bench --locked --no-run --features wasm-bench --package runtime-common
 
-.PHONY: build-release
```
Comment (zjb0807) [inline Makefile:49]: The `run*` commands are also outdated

File: runtime/acala/src/lib.rs [hunk 2]
```diff
@@ -2034,10 +2030,10 @@ mod benches {
 	// frame_benchmarking::define_benchmarks!(
 	// 	// XCM
 	// 	[pallet_xcm, PalletXcmExtrinsicsBenchmark::<Runtime>]
+	// // TODO: add oracle
```
Comment (zjb0807) [inline runtime/acala/src/lib.rs:2033]:
```suggestion
```

File: runtime/karura/src/lib.rs [hunk 3]
```diff
@@ -2062,10 +2062,10 @@ mod benches {
 	// frame_benchmarking::define_benchmarks!(
 	// 	// XCM
 	// 	[pallet_xcm, PalletXcmExtrinsicsBenchmark::<Runtime>]
+	// // TODO: add oracle
```
Comment (zjb0807) [inline runtime/karura/src/lib.rs:2065]:
```suggestion
```

File: runtime/acala/src/lib.rs (hunk 2 above)
Comment (xlc) [reply runtime/acala/src/lib.rs:2033]: the oracle benchmarks are removed in this PR and need to add it back later using the benchmarks in the ORML repo

File: modules/evm-bridge/src/tests.rs [hunk 4]
```diff
@@ -380,33 +381,35 @@ fn tracing_should_work() {
...
+         "type": "CALL",
+         "from": "0x1000000000000000000000000000000000000002",
+         "to": "0x5dddfce53ee040d9eb21afbc0ae1bb4dbb0ba643",
+         "input": "0xa9059cbb0000000000000000000000001000000000000000000000000000000000000001000000000000000000000000000000000000000000000000000000000000000a",
+         "value": "0x0",
+         "gas": 200000,
+         "gasUsed": 200000,
+         "output": null,
+         "error": null,
+         "revertReason": "0xe450d38c00000000000000000000000010000000000000000000000000000000000000020000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000a",
```
Comment (xlc) [inline modules/evm-bridge/src/tests.rs:397]: @zjb0807 revert reason changed. can you confirm if this is ok?

File: ts-tests/tests/test-gas-limit.ts [hunk 5]
```diff
@@ -11,24 +10,24 @@ import { BodhiSigner } from "@acala-network/bodhi";
...
+    beforeAll(async () => {
         [alice] = context.wallets;
     });
 
     it("block gas limit", async () => {
         const contract = await deployContract(alice, Factory);
         // limited by used_storage
-        const result = await contract.createContractLoop(350);
-        expect(result.gasLimit.toNumber()).to.be.eq(3569798122);
+        const result = await contract.createContractLoop(200);
```
Comment (xlc) [inline ts-tests/tests/test-gas-limit.ts:20]: not sure why this is reduced so much. can you check? @zjb0807

File: modules/evm-bridge/src/tests.rs (hunk 4 above)
Comment (zjb0807) [reply modules/evm-bridge/src/tests.rs:397]: https://github.com/OpenZeppelin/openzeppelin-contracts/blob/dc44c9f1a4c3b10af99492eed84f83ed244203f6/contracts/token/ERC20/ERC20.sol#L229

https://github.com/OpenZeppelin/openzeppelin-contracts/blob/dbb6104ce834628e473d2173bbc9d47f81a9eec3/contracts/token/ERC20/ERC20.sol#L195

`openzeppelin-contracts` dependency version has been updated and the restore message has changed

File: ts-tests/tests/test-gas-limit.ts (hunk 5 above)
Comment (zjb0807) [reply ts-tests/tests/test-gas-limit.ts:20]: #2795 fix this

**Review by zjb0807 — APPROVED** [review]

**Pull request merged**